packages/core/src/components/Text.ts

```diff
@@ -60,34 +68,34 @@ export const TuiText = defineComponent({
     onUnmounted(scheduleUpdate)
 
     function transform(text: string): string {
-      if (props.dimmed) {
+      if (propsWithClasses.value.dimmed) {
```

Comment:
reading `.value` has a small overhead, so it's better to read value only once, we could do this:

```ts
const { dimmed, ... } = propsWithClasses.value
```

But instead, let's pass the props as an argument and move the function outside of the component (same file) so we have only one for all component instances